### DISCUSSION TITLE
Renovate only one subdirectory

### DISCUSSION THREAD

**Discussion by cogito-cmurphy** [renovatebot/renovate#10027] [Request Help] [answered]
Hello,

I am new to implementing Renovate and we are currently using this as a proof of concept. I am trying to point renovate to one subdirectory of a repo without having to exclude all other subdirectories individually.

In my renovate.json, I have successfully ignored both "images" and "pipeline", as seen below.
{  "$schema": "https://docs.renovatebot.com/renovate-schema.json",	"ignorePaths": ["images", "pipeline"]

Now the subdirectory I want is base-asdf in v2. 

my v2 dir looks like
v2/base-asdf
v2/base-python
v2/xxx
v2/xxxxx
etc etc

Is there anyway to tell Renovate to ignore all other repos and specifically only look at v2/base-asdf?

Thank you

**Comment by viceice** [comment]
do not use autodiscover and configure [repositories](https://docs.renovatebot.com/self-hosted-configuration/#repositories)

**Reply by cogito-cmurphy** [reply]
I have autodiscover off, and only point to two repos. I used bad wording. I am looking to only use one subdirectory in one of my repositories. See my config in question in the answer to rarkins below.

thank you!

**Comment by rarkins** [comment, marked as answer]
"I want it to run on only one repository" is a different question to "I want it to run on only one subdirectory". Are you asking both questions, or only one?

**Reply by cogito-cmurphy** [reply]
Sorry just one question, one subdirectory. I am going to try @viceice solution. Sorry for the confusion.

**Reply by cogito-cmurphy** [reply]
oh I guess the offer from viceice actually doesn't serve what im trying to do. 
I have a renovate config like
```
    renovate:
      config: |
        {
          "platform": "github",
          "username": "spotig-renovate",
          "endpoint": "https://api.github.com",
          "autodiscover": "false",
          "token": "RENOVATE_TOKEN",
          "dryRun": false,
          "logLevel": "debug",
          "repositories": ["company/k8s-charts", "company/docker-images"],
          "ignoreDeps": [
            "metrics-server",
            "compute-worker",
            "compute-speech-recognition",
            "compute-manager",
            "interaction-computation-service"
          ]
        }
```
    
   This works great, it gets me a check on all files in both k8s-charts and docker-images. However in docker-images there are 3 main directories
   v2/
   images/
   pipelines/
   
   Above I properly ignored images/ & pipelines/ in my renovate.json. However for testing purposes id like to be able to point renovate only at one subdirectory in v2/. Right now it picks up 30+ directories with a lot of changes.
   
   I hope this makes sense.
   
   Thank you!

**Reply by viceice** [reply]
So you should add a package rules to the repo config which simply disables all deps and then you use `matchPaths` rule to reenable only the subdir.

**Reply by cogito-cmurphy** [reply]
thanks ill give this a go

**Reply by cogito-cmurphy** [reply]
I got this to work, which is enough for our POC. Thank you for the assistance @viceice  @rarkins 
```
{  "$schema": "https://docs.renovatebot.com/renovate-schema.json",	"ignorePaths": ["images", "pipeline""includePaths": ["v2/cogito-base-asdf/21.02.25/Dockerfile"]}
```
